nvflare/app_common/hub/hub_controller.py

```diff
@@ -0,0 +1,490 @@
+        start = time.time()
+        while True:
+            if self.run_ended:
+                # tell T1 to end the run
+                self._abort(reason="", abort_signal=abort_signal, fl_ctx=fl_ctx)
+                return
+
+            if abort_signal.triggered:
+                # tell T1 to end the run
+                self._abort(reason="", abort_signal=abort_signal, fl_ctx=fl_ctx)
+                return
+
+            topic, data = self.pipe_monitor.get_next()
+            if not topic:
+                if self.task_wait_time and time.time() - start > self.task_wait_time:
+                    # timed out - tell T1 to end the RUN
+                    self._abort(
+                        reason=f"task data timeout after {self.task_wait_time} secs",
+                        abort_signal=abort_signal,
+                        fl_ctx=fl_ctx,
+                    )
+                    return
+            else:
+                if topic in [Topic.ABORT, Topic.END, Topic.PEER_GONE]:
+                    # the T1 peer is gone
+                    self.log_info(fl_ctx, f"T1 stopped: '{topic}'")
+                    return
+
+                self.log_info(fl_ctx, f"got data for task '{topic}' from T1")
+                if not isinstance(data, Shareable):
+                    self._abort(
+                        reason=f"bad data for task '{topic}' from T1 - must be Shareable but got {type(data)}",
+                        abort_signal=abort_signal,
+                        fl_ctx=fl_ctx,
+                    )
+                    return
+
+                task_name = data.get_header(ReservedHeaderKey.TASK_NAME)
+                if not task_name:
+                    self._abort(
+                        reason=f"bad data for task '{topic}' from T1 - missing task name",
+                        abort_signal=abort_signal,
+                        fl_ctx=fl_ctx,
+                    )
+                    return
+
+                task_id = data.get_header(ReservedHeaderKey.TASK_ID)
+                if not task_id:
+                    self._abort(
+                        reason=f"bad data for task '{topic}' from T1 - missing task id",
+                        abort_signal=abort_signal,
+                        fl_ctx=fl_ctx,
+                    )
+                    return
+
+                op_desc = data.get_header(ReservedHeaderKey.TASK_OPERATOR, {})
+                op_id = op_desc.get(TaskOperatorKey.OP_ID)
+                if not op_id:
+                    op_desc[TaskOperatorKey.OP_ID] = task_name
+
+                self._resolve_op_desc(op_desc)
+                op = op_desc.get(TaskOperatorKey.METHOD, OperatorMethod.BROADCAST)
+                if not op:
+                    self._abort(
+                        reason=f"bad operator in task '{topic}' from T1 - missing op",
+                        abort_signal=abort_signal,
+                        fl_ctx=fl_ctx,
+                    )
+                    return
```

Comment:
The logic of `op = op_desc.get(TaskOperatorKey.METHOD, OperatorMethod.BROADCAST)`
Will return `OperatorMethod.BROADCAST` if op_desc does NOT have key `TaskOperatorKey.METHOD`.

So this if statement will never be evaluated to True.

Maybe we want
```
op = op_desc.get(TaskOperatorKey.METHOD, None)
if op is None:
```